examples/gno.land/p/demo/int256/arithmetic.gno

```diff
@@ -1,202 +1,396 @@
-	if x.neg {
-		if x.abs.Gt(y) {
-			z.abs.Sub(x.abs, y)
-			z.neg = true
-		} else {
-			z.abs.Sub(y, x.abs)
-			z.neg = false
-		}
-	} else {
-		z.abs.Add(x.abs, y)
-		z.neg = false
-	}
+	z.value.Add(&x.value, y)
 	return z
 }
 
-// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
-func AddDelta(z, x *uint256.Uint, y *Int) {
-	if y.neg {
-		z.Sub(x, y.abs)
-	} else {
-		z.Add(x, y.abs)
-	}
-}
-
-// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
-func AddDeltaOverflow(z, x *uint256.Uint, y *Int) bool {
-	var overflow bool
-	if y.neg {
-		_, overflow = z.SubOverflow(x, y.abs)
-	} else {
-		_, overflow = z.AddOverflow(x, y.abs)
-	}
-	return overflow
-}
-
-// Sub sets z to the difference x-y and returns z.
+// Sub subtracts two int256 values and saves the result in z.
 func (z *Int) Sub(x, y *Int) *Int {
-	z.initiateAbs()
-
-	if x.neg != y.neg {
-		// If sign are different, add the absolute values
-		z.abs.Add(x.abs, y.abs)
-		z.neg = x.neg
-	} else {
-		switch x.abs.Cmp(y.abs) {
-		case 1: // x > y
-			z.abs.Sub(x.abs, y.abs)
-			z.neg = x.neg
-		case -1: // x < y
-			z.abs.Sub(y.abs, x.abs)
-			z.neg = !x.neg
-		case 0: // x == y
-			z.abs = uint256.NewUint(0)
-		}
-	}
-
-	// Ensure zero is always positive
-	if z.abs.IsZero() {
-		z.neg = false
-	}
+	z.value.Sub(&x.value, &y.value)
 	return z
 }
 
-// SubUint256 set z to the difference x - y, where y is a uint256, and returns z
+// SubUint256 subtracts uint256 and int256 values and saves the result in z.
 func (z *Int) SubUint256(x *Int, y *uint256.Uint) *Int {
-	if x.neg {
-		z.abs.Add(x.abs, y)
-		z.neg = true
-	} else {
-		if x.abs.Lt(y) {
-			z.abs.Sub(y, x.abs)
-			z.neg = true
-		} else {
-			z.abs.Sub(x.abs, y)
-			z.neg = false
-		}
-	}
+	z.value.Sub(&x.value, y)
 	return z
 }
 
-// Mul sets z to the product x*y and returns z.
+// Mul multiplies two int256 values and saves the result in z.
+//
+// It considers the signs of the operands to determine the sign of the result.
 func (z *Int) Mul(x, y *Int) *Int {
-	z.initiateAbs()
+	xAbs, xSign := x.Abs(), x.Sign()
+	yAbs, ySign := y.Abs(), y.Sign()
+
+	z.value.Mul(xAbs, yAbs)
+
+	if xSign != ySign {
+		z.value.Neg(&z.value)
+	}
 
-	z.abs = z.abs.Mul(x.abs, y.abs)
-	z.neg = x.neg != y.neg && !z.abs.IsZero() // 0 has no sign
 	return z
 }
 
-// MulUint256 sets z to the product x*y, where y is a uint256, and returns z
-func (z *Int) MulUint256(x *Int, y *uint256.Uint) *Int {
-	z.abs.Mul(x.abs, y)
-	if z.abs.IsZero() {
-		z.neg = false
-	} else {
-		z.neg = x.neg
+// Abs returns the absolute value of z.
+func (z *Int) Abs() *uint256.Uint {
+	if z.Sign() >= 0 {
+		return &z.value
 	}
-	return z
+
+	var absValue uint256.Uint
+	absValue.Sub(uint0, &z.value).Neg(&z.value)
+
+	return &absValue
 }
 
-// Div sets z to the quotient x/y for y != 0 and returns z.
-func (z *Int) Div(x, y *Int) *Int {
-	z.initiateAbs()
+/* +--------------------------------------------------------------------------+
```

Comment:
Can this be package documentation? (So, gnodoc before the `package` clause)

Reply:
moved https://github.com/gnolang/gno/pull/2846/commits/05d3c16ac44dcc30c2564bef68c27398472cfc95